launcher/minecraft/mod/ResourceFolderModel.h

```diff
@@ -125,6 +125,8 @@ class ResourceFolderModel : public QAbstractListModel {
         [[nodiscard]] bool lessThan(const QModelIndex& source_left, const QModelIndex& source_right) const override;
     };
 
+    QString instDirPath() const;
```

Comment:
maybe just make this a parameter for the constructor of all models instead.

Reply:
I'm not actually sure what you mean here?

Reply:
I think he means that, instead of hardcoding the relative path to the instance dir path, as you have done you, you can instead make `ResourceFolderModel`s constructor receive a `QDir` that's the instance directory, and store it in the class.

You can also, instead of getting a `QDir`, just get a `BaseInstance` reference itself. This may future-proof some stuff :p

Reply:
Ok, that makes sense. the modals now take a shared pointer to a BaseInstance.
This function uses `m_instance->instanceRoot()`